### PR TITLE
Add email login warning

### LINKED ISSUE #1295: If user uses an @ in the username warn him to not use his email to login

### PR DESCRIPTION
**Pull request by Katharsas**
Adds new warning if user tries to login with username looking like email
Fixes #1295

### CODE REVIEW

File: src/main/java/com/faforever/client/login/LoginController.java [hunk 1]
```diff
@@ -44,11 +46,13 @@
 @Slf4j
 public class LoginController implements Controller<Node> {
 
+  private static final Pattern emailRegex = Pattern.compile(".*[@].*[.].*");
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/login/LoginController.java:49]: constants go in UPPERCASE_WITH_UNDERSCORES in our codestyle

File: src/test/java/com/faforever/client/login/LoginControllerTest.java [hunk 2]
```diff
@@ -76,4 +85,15 @@ public void testForgotPasswordButtton() throws Exception {
...
+
+  @Test
+  public void testUsernameEmailWarning() {
+    instance.usernameInput.setText("test@example.com");
+    instance.passwordInput.setText("foo");
+    instance.loginButton.fire();
+    verify(i18n).get(LOGIN_WITH_EMAIL_WARNING_KEY);
+    verifyZeroInteractions(userService);
+    assertThat(instance.loginErrorLabel.isVisible(), is(true));
+    assertThat(instance.loginErrorLabel.getText(), is(LOGIN_WITH_EMAIL_WARNING_KEY));
```
Comment (Brutus5000) [inline src/test/java/com/faforever/client/login/LoginControllerTest.java:97]: In positive case we should assert that the loginErrorLabel is not visible.

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: src/test/java/com/faforever/client/login/LoginControllerTest.java (hunk 2 above)
Comment (1-alex98) [reply src/test/java/com/faforever/client/login/LoginControllerTest.java:97]: Well this not really his job tho

Comment (Brutus5000) [reply src/test/java/com/faforever/client/login/LoginControllerTest.java:97]: What? If the code is wrong and always shows the error message? how do you test it then?

**Pull request merged**